src/code_08005e54.c

```diff
@@ -1,31 +1,228 @@
 #include "code_08005e54.h"
+#include "memory_heap.h"
+#include "code_08001360.h"
 
-asm(".include \"include/gba.inc\"");//Temporary
 
 /* ? */
 
-#include "asm/code_08005e54/asm_08005e54.s"
+u32 func_08005e54(struct struct_08005e54 *arg0, s32 *arg1, s32 *arg2, s32 *arg3, s32 *arg4) {
+    s32 temp_r2;
+    s32 temp_ip;
 
-#include "asm/code_08005e54/asm_08005ee4.s"
+    temp_r2 = *arg1 + *arg3;
+    temp_ip = *arg2 + *arg4;
 
-#include "asm/code_08005e54/asm_08005f04.s"
+    if (((*arg1 >= arg0->unk4) || (temp_r2 <= 0)) || (*arg2 >= arg0->unk6) || (temp_ip <= 0)) {
+        *arg4 = 0;
+        *arg3 = 0;
+        return 0;
+    }
+    if (*arg1 < 0) {
+        *arg3 = temp_r2;
+        *arg1 = 0;
+    }
+    if (temp_r2 > arg0->unk4) {
+        *arg3 -= temp_r2 - arg0->unk4;
+    }
+    if (*arg2 < 0) {
+        *arg4 += *arg2;
+        *arg2 = 0;
+    }
+    if (temp_ip > arg0->unk6) {
+        *arg4 -= temp_ip - arg0->unk6;
+    }
+    return *arg3 * *arg4;
+}
 
-#include "asm/code_08005e54/asm_08005f34.s"
+void func_08005ee4(struct struct_08005e54 *arg0, u16 arg1) {
+    func_08005f34(arg0, 0, 0, arg0->unk4, arg0->unk6, arg1);
+}
 
-#include "asm/code_08005e54/asm_08005fe4.s"
+void func_08005f04(struct struct_08005e54 *arg0, s32 arg1, s32 arg2, u16 arg3) {
+    if ((arg1 >= 0) && (arg1 < arg0->unk4) && (arg2 >= 0) && (arg2 < arg0->unk6)) {
+        arg0->unk0[arg1 + arg0->unk4 * arg2] = arg3;
+    }
+}
 
-#include "asm/code_08005e54/asm_080060bc.s"
+void func_08005f34(struct struct_08005e54 *arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, u16 arg5) {
+    u16 *temp_r2;
+    u32 i;
+    
+    if ((arg3 <= 0) || (arg4 <= 0)) {
+        return;
+    }
+    if (func_08005e54(arg0, &arg1, &arg2, &arg3, &arg4) == 0) {
+        return;
+    }
+    temp_r2 = &arg0->unk0[arg1] + arg2 * arg0->unk4; 
+    if (arg3 > 1) {
+        while (arg4-- != 0) {
+            for (i = arg3; i >= 8; temp_r2 += 8, i -= 8) {
+                temp_r2[0] = temp_r2[1] = temp_r2[2] = temp_r2[3] = temp_r2[4] = temp_r2[5] = temp_r2[6] = temp_r2[7] = arg5;
```

Comment:
is this necessary? cant these be written as separate lines

Reply:
https://decomp.me/scratch/VB9rr doesn't feel like something a real programmer would write, IMHO